### PR TITLE
feat: add join_lpr3 with tests

### PR DESCRIPTION
**Pull request by signekb**
## Description

- This PR adds the function `join_lpr3()` with tests as a stacked PR to #88 
- It's largely based on #88 , joining the LPR3 registers `diagnoser` and `kontakter` with a full join.

### CODE REVIEW

File: tests/testthat/test-joins.R [hunk 1]
```diff
@@ -56,3 +58,66 @@ test_that("joining works for data.table", {
...
   expect_contains(class(actual), "data.table")
 })
+
+# join_lpr3 -----------------------------------------------------------------
+
+actual_diagnoser <- tibble::tibble(
+  dw_ek_kontakt = 1:2,
+  diagnosekode = c("DA071","DD075"),
+  diagnosetype = c("A", "B"),
+  senere_afkraefter = c("Nej", "Ja")
```
Comment (signekb) [inline tests/testthat/test-joins.R:68]:
```suggestion
  senere_afkraeftet = c("Nej", "Ja")
```

File: tests/testthat/test-joins.R [hunk 2]
```diff
@@ -56,3 +58,66 @@ test_that("joining works for data.table", {
...
+)
+
+actual_kontakter <- tibble::tibble(
+  cpr = c(1, 1, 2),
+  dw_ek_kontakt = 1:3,
+  dato_start = c("20230101", "20220101", "20200101"),
+  hovedspaciale_ans = c("Neurologi", "Akut medicin", "Kardiologi"),
```
Comment (signekb) [inline tests/testthat/test-joins.R:75]:
```suggestion
  hovedspeciale_ans = c("Neurologi", "Akut medicin", "Kardiologi"),
```

File: tests/testthat/test-joins.R [hunk 3]
```diff
@@ -56,3 +58,66 @@ test_that("joining works for data.table", {
...
+)
+
+expected_lpr3 <- tibble::tibble(
+  cpr = c(1, 1, 2),
+  dw_ek_kontakt = c(1,2,3),
+  dato_start = c("20230101", "20220101", "20200101"),
+  hovedspaciale_ans = c("Neurologi", "Akut medicin", "Kardiologi"),
```
Comment (signekb) [inline tests/testthat/test-joins.R:82]:
```suggestion
  hovedspeciale_ans = c("Neurologi", "Akut medicin", "Kardiologi"),
```

File: tests/testthat/test-joins.R [hunk 4]
```diff
@@ -56,3 +58,66 @@ test_that("joining works for data.table", {
...
+  diagnosekode = c("DA071","DD075", NA),
+  diagnosetype = c("A", "B", NA),
+  senere_afkraefter = c("Nej", "Ja", NA),
```
Comment (signekb) [inline tests/testthat/test-joins.R:85]:
```suggestion
  senere_afkraeftet = c("Nej", "Ja", NA),
```

**Review by lwjohnst86 — APPROVED** [review]
Merged from main, resolved conflicts, revise tests and join to match the join_lpr2 that we updated.

**Comment by signekb** [comment]
Nice! Thanks, @lwjohnst86 👍

**Review by Aastedet — APPROVED** [review]

**Pull request merged**